### ISSUE TITLE
Provide "guest" user type

### ISSUE THREAD

**Issue by RocketRaccoon** [reportportal/reportportal#2]
Hi,

Please provide "guest" account which can view data without logging in (like in Jenkins, Upsource, etc.).
For example such account can view only shared filters and dashboards.

**Comment by DzmitryHumianiuk** [comment]
@RocketRaccoon 
could you please describe basis flow for this user role?

**Comment by DzmitryHumianiuk** [comment]
Will be considered with PostgreSQL migration.

authorization service will be replaced with simple mock, which will always return only one user

**Comment by evjlobanova** [comment]
This issue includes #164 , #166 , #261 .

**Comment by DzmitryHumianiuk** [comment]
cc @tareqalayan

**Comment by DzmitryHumianiuk** [comment]
#396 related and mostly equal request

**Comment by evjlobanova** [comment]
In issue  #583 there is described the approach that most likely will be used for this problem. 

We can add an auto-role Guest on the project with Watch only access to ReportPortal.
The dashboard link will be included credentials of guest user.
If somebody shares a link with the non-authorized user, the credentials of the guest will be added automatically.
A non-authorized user will be seeing just dashboards and launches in Watch only mode.
If the authorized user will click on the link, he will see info in accordance with his permissions.

**Comment by balukov** [comment]
It is hard to use full features RP without this feature.
Cuz all who want to see the report  (developers, managers) should have credentials.
And this is the blocker for us.
You know developers - they want just click and see results. 
If they need to set a login/password every time - this makes smaller their desire to see the report.

**Comment by evjlobanova** [comment]
@balukov yes, you are absolutely right, we are planning to make this feature. Unfortunately, I can not tell you a solid ETA for this feature.
For now, you can try to use login with GitHub/LDAP or SAML, it can help to reduce the time for achieving results.

If this feature is crucial for you, we also can provide paid support. In the scope of the paid contract, we can gather requirements with you and implement it.

**Comment by nbleiberg** [comment]
I wonder if an alternate/additional solution could be to provide dashboards within email notifications.
This would allow a team to create a mailing list address that will be used to create a single user in RP and sign that address up for that kind of email notification with any dashboards they want. Then anyone can join that mailing list to receive those dashboards.

**Comment by evjlobanova** [comment]
Yes, we are planning to use a similar approach in order to provide a possibility to share a link to dashboard and also to use embedded widgets.

**Comment by avikghosh** [comment]
Hey @evjlobanova any update on this?

**Comment by diogormendes** [comment]
Looking forward to an Update as well!

**Comment by NishKebab** [comment]
Any update on this?

**Comment by DzmitryHumianiuk** [comment]
@macpijan not yet

**Comment by DzmitryHumianiuk** [comment]
@RocketRaccoon Hah )
Yes, it looks silly, but helps us to track features.
We still have this one on a big roadmap, and even plan to make it available in community version.

Now big bets are placed on version 6, with a fundamental changes in data structure and reporting principles.
The inability to make embedded widgets from the RP to other systems was due to the limitations of the authorization mechanisms in the current versions. At the moment, we are designing the architecture of version 6, and we are putting this capability in right away. The presence of such an opportunity will make the implementation of this feature simple and fast. But working on the architecture takes time.

Version 6 is planned to become the crown of our experience and knowledge, and to incorporate the revised requests from the community over the past 4 years. Everything that users asked us to add to the app, now we will make it possible to be added. And we take this into account in the big picture, and build immediately taking into account all these requests.

Plugin system will also play its' role. When an added functionality can be loaded just as a binary file right on the go.
Which will open up a new way for community to contribute into the application.

**Comment by dgur** [comment]
Amazing! Thank you for the detailed explanation and road map!

**Comment by macpijan** [comment]
I'm looking forward to this version then ;)

Unless maybe there is another way of sharing the tests results publicly for open-source projects - which is something I'm interested in. I'm looking through the available tools at the moment, and ReportPortal was really appealing from all of the other ways so far, but maybe it is not the tool meant for that usecase? Apologies for slight off-topic.

**Comment by sanketshashank** [comment]
Any estimate to add this feature in nearby soon ?

**Comment by macpijan** [comment]
Just curious whether there was any update on this in the last couple of months?

**Comment by DzmitryHumianiuk** [comment]
@macpijan unfortunately not.

**Comment by hunteryinAS** [comment]
I think this feature is important in QA world. Just want to check if any update after another year :)

**Comment by DzmitryHumianiuk** [comment]
@hunteryinAS Still no, please check again next year.

**Comment by bryansancheZ199** [comment]
Hello, I am also looking for this functionality. Is there any update for this or can anyone please tell me some other software can I use to show launches results?
